src/JBMultiTerminal.sol

```diff
@@ -406,15 +406,15 @@ contract JBMultiTerminal is JBPermissioned, ERC2771Context, IJBMultiTerminal {
                 knownInvalidDecimals = true;
             } else if (accountingContext.token != JBConstants.NATIVE_TOKEN) {
                 // slither-disable-next-line calls-loop
-                try IERC165(accountingContext.token).supportsInterface(type(IERC20Metadata).interfaceId) returns (
-                    bool doesSupport
-                ) {
+                try IERC20Metadata(accountingContext.token).decimals() returns (uint8 decimals) {
                     // slither-disable-next-line calls-loop
-                    if (doesSupport && accountingContext.decimals != IERC20Metadata(accountingContext.token).decimals())
-                    {
+                    if (accountingContext.decimals != decimals) {
                         knownInvalidDecimals = true;
                     }
-                } catch {}
+                } catch {
+                    // The token didn't support `decimals`.
+                    knownInvalidDecimals = true;
```

Comment:
This line has the side-effect that tokens that do not implement `decimals()` would no longer be supported, where before we did support them and assumed the user must have given us the correct `decimals`. 

Is this expected behavior or should we allow tokens where `decimals()` is not supported?

Reply:
good question. lets allow user input if there's no `decimals()`